File: Backend/AstroGame.Shared/Models/Researches/OneTimeResearchCost.cs
using AstroGame.Shared.Models.Technologies;

namespace AstroGame.Shared.Models.Researches
{
    public class OneTimeResearchCost : TechnologyCost
    {
        public double Amount { get; set; }

        public virtual OneTimeResearch OneTimeResearch { get; set; }
    }
}

File: Frontend/apps/astrogame_app/lib/models/researches/one_time_research_cost.dart
import 'package:astrogame_app/communications/converters/guid_converter.dart';
import 'package:astrogame_app/models/technologies/technology_cost.dart';
import 'package:json_annotation/json_annotation.dart';

part 'one_time_research_cost.g.dart';

@GuidConverter()
@JsonSerializable()
class OneTimeResearchCost extends TechnologyCost {
  double amount;

  OneTimeResearchCost();

  factory OneTimeResearchCost.fromJson(Map<String, dynamic> json) =>
      _$OneTimeResearchCostFromJson(json);
  Map<String, dynamic> toJson() => _$OneTimeResearchCostToJson(this);
}
